packages/framework-provider-azure/test/helpers/query-helper.test.ts

```diff
@@ -169,6 +169,46 @@ describe('Query helper', () => {
       )
     })
 
+    it('Executes a SQL query with a projectionFor projection that has a deeply nested array of objects', async () => {
+      await search(
+        mockCosmosDbClient as any,
+        mockConfig,
+        mockReadModelName,
+        {},
+        undefined,
+        undefined,
+        false,
+        undefined,
+        [
+          'id',
+          'x.arr[].z',
+          'foo.bar.items[].id',
+          'foo.bar.baz.items[].id',
+          'arr[].subArr[].id',
+          'arr[].id',
+        ] as ProjectionFor<unknown>
+      )
+
+      expect(mockCosmosDbClient.database).to.have.been.calledWithExactly(mockConfig.resourceNames.applicationStack)
+      expect(
+        mockCosmosDbClient.database(mockConfig.resourceNames.applicationStack).container
+      ).to.have.been.calledWithExactly(`${mockReadModelName}`)
+      expect(
+        mockCosmosDbClient.database(mockConfig.resourceNames.applicationStack).container(`${mockReadModelName}`).items
+          .query
+      ).to.have.been.calledWith(
+        match({
+          query:
+            'SELECT c["id"], ' +
+            'ARRAY(SELECT item["z"] FROM item IN c["x"]["arr"]) AS "x.arr", ' +
+            'c["foo"]["bar"]["items"] AS "foo.bar.items", c["foo"]["bar"]["baz"]["items"] AS "foo.bar.baz.items", ' +
+            'ARRAY(SELECT item["subArr"], item["id"] FROM item IN c["arr"]) AS arr ' +
+            'FROM c ',
```

Comment:
You may want to use backticks ``` instead for this multiline string, as it is explained https://eranstiller.com/javascript-multiline-strings

```typescript
query: `
SELECT c["id"], 
ARRAY(SELECT item["z"] FROM item IN c["x"]["arr"]) AS "x.arr", 
c["foo"]["bar"]["items"] AS "foo.bar.items", c["foo"]["bar"]["baz"]["items"] AS "foo.bar.baz.items", 
ARRAY(SELECT item["subArr"], item["id"] FROM item IN c["arr"]) AS arr 
FROM c
`
```

Reply:
The problem with the backticks is that they add newline characters that are not in the generated string we're comparing it to. Therefore, we have to use string concatenation. Otherwise, the tests fail.